dbloader/riak_loader.py

```diff
@@ -127,25 +127,19 @@ def truncate(self, bucket_name, table=None, custom=None):
         tbucket = self.conn.bucket(bucket_name)
         keycount = 0
         get_one = 0
-        kfile = self.keyfile + ".{0}".format(bucket_name)
-        with open(kfile, "w") as myFile:
-            try:
-                with closing(self.conn.stream_keys(tbucket, self.timeout)) as keys:
-                    for key_list in keys:
-                        for key in key_list:
-                            keycount = keycount + 1
-                            myFile.write("%s\n" % key)
-            except Exception as e:
-                logger.error("Unknown exception for %s (%s)", bucket_name, e)
-                pass
+        deletes = []
+        try:
+            tpool = Pool(self.concurrency)
+            with closing(self.conn.stream_keys(tbucket, self.timeout)) as keys:
+                for key in keys:
+                    keycount = keycount + 1
+                    deletes.append(tpool.spawn(self.delete, bucket_name, None, key.strip()))
```

Comment:
Just verifying, previously it was two for loops, do we not need it? :
```
with closing(self.conn.stream_keys(tbucket, self.timeout)) as keys:
                    for key_list in keys:
                        for key in key_list:
                            keycount = keycount + 1
```

Reply:
Correct - the two loops were replaced with one because we are now skipping the `write keys to file` step.

I validated it against staging riak as well